<a href="https://colab.research.google.com/github/aditi10998/EDA_Captsone_Project_2/blob/main/Copy_of_Rossmann_Sales_Prediction_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# <b><u> Project Title : Sales Prediction : Predicting sales of a major store chain Rossmann</u></b>

## <b> Problem Description </b>

### Rossmann operates over 3,000 drug stores in 7 European countries. Currently, Rossmann store managers are tasked with predicting their daily sales for up to six weeks in advance. Store sales are influenced by many factors, including promotions, competition, school and state holidays, seasonality, and locality. With thousands of individual managers predicting sales based on their unique circumstances, the accuracy of results can be quite varied.

### You are provided with historical sales data for 1,115 Rossmann stores. The task is to forecast the "Sales" column for the test set. Note that some stores in the dataset were temporarily closed for refurbishment.

## <b> Data Description </b>

### <b>Rossmann Stores Data.csv </b> - historical data including Sales
### <b>store.csv </b> - supplemental information about the stores


### <b><u>Data fields</u></b>
### Most of the fields are self-explanatory. The following are descriptions for those that aren't.

* #### Id - an Id that represents a (Store, Date) duple within the test set
* #### Store - a unique Id for each store
* #### Sales - the turnover for any given day (this is what you are predicting)
* #### Customers - the number of customers on a given day
* #### Open - an indicator for whether the store was open: 0 = closed, 1 = open
* #### StateHoliday - indicates a state holiday. Normally all stores, with few exceptions, are closed on state holidays. Note that all schools are closed on public holidays and weekends. a = public holiday, b = Easter holiday, c = Christmas, 0 = None
* #### SchoolHoliday - indicates if the (Store, Date) was affected by the closure of public schools
* #### StoreType - differentiates between 4 different store models: a, b, c, d
* #### Assortment - describes an assortment level: a = basic, b = extra, c = extended
* #### CompetitionDistance - distance in meters to the nearest competitor store
* #### CompetitionOpenSince[Month/Year] - gives the approximate year and month of the time the nearest competitor was opened
* #### Promo - indicates whether a store is running a promo on that day
* #### Promo2 - Promo2 is a continuing and consecutive promotion for some stores: 0 = store is not participating, 1 = store is participating
* #### Promo2Since[Year/Week] - describes the year and calendar week when the store started participating in Promo2
* #### PromoInterval - describes the consecutive intervals Promo2 is started, naming the months the promotion is started anew. E.g. "Feb,May,Aug,Nov" means each round starts in February, May, August, November of any given year for that store

In this Capstone Project.We are going to going to build a model using regression in Machine learning for forecasting the "Sales" column for the test set.
Let's first understand what is regression technique.
Regression analysis is a statistical method to model the relationship between a dependent (target) and independent (predictor) variables with one or more independent variables. More specifically, Regression analysis helps us to understand how the value of the dependent variable is changing corresponding to an independent variable when other independent variables are held fixed. It predicts continuous/real values such as temperature, age, salary, price, etc.

---




In [ ]:
#importing necessary libraries
import pandas as pd

We will import other libraries as per our requirement.

In [ ]:
#Mounting Google drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


To begin we will first need to explore our dataset.
Let's import our csv files and convert into dataframes.
Here we have two datasets Rossmann Stores Data and stores which consist of information of stores and stores related aspects.

In [ ]:
File1="/content/drive/MyDrive/EDA_Capstone_Project2/Data & Resources/Copy of Rossmann Stores Data.csv"
File2="/content/drive/MyDrive/EDA_Capstone_Project2/Data & Resources/Copy of store.csv"

In [ ]:
#Loading the files
Rossmann_Stores_df=pd.read_csv(File1)
Store_df=pd.read_csv(File2)

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (7) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


Let's perform simple tasks on our dataset and explore it more by usng different methods.

In [ ]:
#To display total number of rows and columns
Rossmann_Stores_df.shape

(1017209, 9)

In this dataset we can see rows are 1017209 and columns are 9.In all it is very huge data.

In [ ]:
#to display total number of rows and columns
Store_df.shape

(1115, 10)

In this dataset there are 1115 rows and 10 columns.As compared to previous dataset this is not that large amount of data.




In [ ]:
#To display names of column ,thier datatypes and missing values
Rossmann_Stores_df.info()
Store_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1017209 entries, 0 to 1017208
Data columns (total 9 columns):
 #   Column         Non-Null Count    Dtype 
---  ------         --------------    ----- 
 0   Store          1017209 non-null  int64 
 1   DayOfWeek      1017209 non-null  int64 
 2   Date           1017209 non-null  object
 3   Sales          1017209 non-null  int64 
 4   Customers      1017209 non-null  int64 
 5   Open           1017209 non-null  int64 
 6   Promo          1017209 non-null  int64 
 7   StateHoliday   1017209 non-null  object
 8   SchoolHoliday  1017209 non-null  int64 
dtypes: int64(7), object(2)
memory usage: 69.8+ MB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1115 entries, 0 to 1114
Data columns (total 10 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Store                      1115 non-null   int64  
 1   StoreType                  1115 non-null   object 
 2   Assort

In [ ]:
#To overview statistics approach of numerical variables
Rossmann_Stores_df.describe()


Store     DayOfWeek  ...         Promo  SchoolHoliday
count  1.017209e+06  1.017209e+06  ...  1.017209e+06   1.017209e+06
mean   5.584297e+02  3.998341e+00  ...  3.815145e-01   1.786467e-01
std    3.219087e+02  1.997391e+00  ...  4.857586e-01   3.830564e-01
min    1.000000e+00  1.000000e+00  ...  0.000000e+00   0.000000e+00
25%    2.800000e+02  2.000000e+00  ...  0.000000e+00   0.000000e+00
50%    5.580000e+02  4.000000e+00  ...  0.000000e+00   0.000000e+00
75%    8.380000e+02  6.000000e+00  ...  1.000000e+00   0.000000e+00
max    1.115000e+03  7.000000e+00  ...  1.000000e+00   1.000000e+00

[8 rows x 7 columns]

In [ ]:
Store_df.describe()

Store  CompetitionDistance  ...  Promo2SinceWeek  Promo2SinceYear
count  1115.00000          1112.000000  ...       571.000000       571.000000
mean    558.00000          5404.901079  ...        23.595447      2011.763573
std     322.01708          7663.174720  ...        14.141984         1.674935
min       1.00000            20.000000  ...         1.000000      2009.000000
25%     279.50000           717.500000  ...        13.000000      2011.000000
50%     558.00000          2325.000000  ...        22.000000      2012.000000
75%     836.50000          6882.500000  ...        37.000000      2013.000000
max    1115.00000         75860.000000  ...        50.000000      2015.000000

[8 rows x 7 columns]

In [ ]:
#To display first n number of rows 
Rossmann_Stores_df.head(10)

Store  DayOfWeek        Date  Sales  ...  Open  Promo  StateHoliday SchoolHoliday
0      1          5  2015-07-31   5263  ...     1      1             0             1
1      2          5  2015-07-31   6064  ...     1      1             0             1
2      3          5  2015-07-31   8314  ...     1      1             0             1
3      4          5  2015-07-31  13995  ...     1      1             0             1
4      5          5  2015-07-31   4822  ...     1      1             0             1
5      6          5  2015-07-31   5651  ...     1      1             0             1
6      7          5  2015-07-31  15344  ...     1      1             0             1
7      8          5  2015-07-31   8492  ...     1      1             0             1
8      9          5  2015-07-31   8565  ...     1      1             0             1
9     10          5  2015-07-31   7185  ...     1      1             0             1

[10 rows x 9 columns]

In [ ]:
Store_df.head(10)

Store StoreType  ... Promo2SinceYear    PromoInterval
0      1         c  ...             NaN              NaN
1      2         a  ...          2010.0  Jan,Apr,Jul,Oct
2      3         a  ...          2011.0  Jan,Apr,Jul,Oct
3      4         c  ...             NaN              NaN
4      5         a  ...             NaN              NaN
5      6         a  ...             NaN              NaN
6      7         a  ...             NaN              NaN
7      8         a  ...             NaN              NaN
8      9         a  ...             NaN              NaN
9     10         a  ...             NaN              NaN

[10 rows x 10 columns]

In [ ]:
#To display last n number of rows 
Rossmann_Stores_df.tail(10)

Store  DayOfWeek        Date  ...  Promo  StateHoliday  SchoolHoliday
1017199   1106          2  2013-01-01  ...      0             a              1
1017200   1107          2  2013-01-01  ...      0             a              1
1017201   1108          2  2013-01-01  ...      0             a              1
1017202   1109          2  2013-01-01  ...      0             a              1
1017203   1110          2  2013-01-01  ...      0             a              1
1017204   1111          2  2013-01-01  ...      0             a              1
1017205   1112          2  2013-01-01  ...      0             a              1
1017206   1113          2  2013-01-01  ...      0             a              1
1017207   1114          2  2013-01-01  ...      0             a              1
1017208   1115          2  2013-01-01  ...      0             a              1

[10 rows x 9 columns]

In [ ]:
Store_df.tail(10)

Store StoreType  ... Promo2SinceYear     PromoInterval
1105   1106         a  ...          2013.0   Jan,Apr,Jul,Oct
1106   1107         a  ...          2010.0   Jan,Apr,Jul,Oct
1107   1108         a  ...             NaN               NaN
1108   1109         c  ...          2012.0   Jan,Apr,Jul,Oct
1109   1110         c  ...             NaN               NaN
1110   1111         a  ...          2013.0   Jan,Apr,Jul,Oct
1111   1112         c  ...             NaN               NaN
1112   1113         a  ...             NaN               NaN
1113   1114         a  ...             NaN               NaN
1114   1115         d  ...          2012.0  Mar,Jun,Sept,Dec

[10 rows x 10 columns]